### ISSUE TITLE
Normalization

### ISSUE THREAD

**Issue by Ravby** [MOEAFramework/MOEAFramework#21]
I have one question about normalization. Are some of the refrence Pareto sets already normalized? For instance the refrence sets for ZDT1-ZDT6 have values in the range [0,1]. This means that the max value is 1 and min value is 0. So when I normalize my approximation set with the refrence set then the normalized approximation set won't be in the range [0,1] if there are values bigger than 1. But if I normalize the approximation with it's own min and max values then I can't compare it to the refrence set.

**Comment by dhadka** [comment]
Hello.  The Pareto sets found in the pf/ folder are not normalized.  They are normalized prior to computing the performance indicators.  The steps are:

1. Load the reference set
2. Compute the min and max bounds for each objective in the reference set
3. Normalize the reference set with these bounds
4. Then, for a given approximation set, normalize the approximation set given the bounds
5. Compute the metric (e.g., GD is computed as the average distance of the normalized approximation set to the normalized reference set)

**Comment by Ravby** [comment]
Thank you for the response. So it is just a coincidence that the true Pareto for the problems ZDT1-ZDT6 has values in the range [0,1].

**Comment by kalyanmoydeb** [comment]
You can always modify ZDT problems to have a different scaling of objective values. For example, f1(x) of ZDT1 can be modified as f1(x)=1000*x1 and f2(x) can be kept as it is, f2(x)=g(x)*(1-sqrt(x1/g(x))) and all variables are bounded within [0,1]. If objective-space normalization is an issue for an algorithm, above modification will test it well, but if variable-space normalization is an issue, then above modification will not be able to test. We can then modify as follows:
f1(x) = x1, f2(x)=g(x)*(1-sqrt(x1/(1000*g(x))))
and x1 is bounded within [0,1000] and all other variables within [0,1].

Hope this helps. 
Kalyan

**Comment by dhadka** [comment]
@Ravby 

Yes, it's just a coincidence that the ZDT problems are bounded by [0, 1].  You can find some visuals and equations for the Pareto sets here: http://www.tik.ee.ethz.ch/sop/download/supplementary/testproblems/

To add to Kalyan's suggestion, here is some example code that applies an additional scaling factor to each objective for a given problem: https://gist.github.com/dhadka/103bf734ed5f3c744759

Best Regards
